bin/stress-test/src/main.rs

```diff
@@ -0,0 +1,444 @@
+fn create_consume_note_txs(
+    block_ref: &BlockHeader,
+    accounts: Vec<Account>,
+    notes: Vec<Note>,
+    note_proofs: &BTreeMap<NoteId, NoteInclusionProof>,
+) -> Vec<ProvenTransaction> {
+    accounts
+        .into_iter()
+        .zip(notes)
+        .map(|(mut account, note)| {
+            let inclusion_proof = note_proofs.get(&note.id()).unwrap();
+            create_consume_note_tx(
+                block_ref,
+                &mut account,
+                InputNote::authenticated(note, inclusion_proof.clone()),
+            )
+        })
+        .collect()
+}
+
+/// Creates a transaction that creates an account and consumes the given input note.
+/// The account is updated with the assets from the input note, and the nonce is set to 1.
+fn create_consume_note_tx(
+    block_ref: &BlockHeader,
+    account: &mut Account,
+    input_note: InputNote,
+) -> ProvenTransaction {
+    let init_hash = account.init_hash();
+
+    input_note.note().assets().iter().for_each(|asset| {
+        account.vault_mut().add_asset(*asset).unwrap();
+    });
+    account.set_nonce(Felt::ONE).unwrap();
+
+    ProvenTransactionBuilder::new(
+        account.id(),
+        init_hash,
+        account.hash(),
+        block_ref.block_num(),
+        block_ref.hash(),
+        u32::MAX.into(),
+        ExecutionProof::new(Proof::new_dummy(), HashFunction::default()),
+    )
+    .add_input_notes(vec![input_note])
+    .build()
+    .unwrap()
+}
+
+/// Creates a transaction from the faucet that creates the given output notes.
+/// Updates the faucet account to increase the issuance slot and it's nonce.
+fn create_emit_note_tx(
+    block_ref: &BlockHeader,
+    faucet: &mut Account,
+    output_notes: Vec<Note>,
+) -> ProvenTransaction {
+    let initial_account_hash = faucet.hash();
+
+    let slot = faucet.storage().get_item(2).unwrap();
+    faucet
+        .storage_mut()
+        .set_item(0, [slot[0], slot[1], slot[2], slot[3] + Felt::new(10)])
+        .unwrap();
+    faucet.set_nonce(faucet.nonce() + Felt::ONE).unwrap();
+
+    ProvenTransactionBuilder::new(
+        faucet.id(),
+        initial_account_hash,
+        faucet.hash(),
+        block_ref.block_num(),
+        block_ref.hash(),
+        u32::MAX.into(),
+        ExecutionProof::new(Proof::new_dummy(), HashFunction::default()),
+    )
+    .add_output_notes(output_notes.into_iter().map(OutputNote::Full).collect::<Vec<OutputNote>>())
+    .build()
+    .unwrap()
+}
+
+/// Get the batch inputs from the store and track the query time on the metrics.
+async fn get_batch_inputs(
+    store_client: &StoreClient,
+    block_ref: &BlockHeader,
+    notes: &[Note],
+    metrics: &mut Metrics,
+) -> BatchInputs {
+    let start = Instant::now();
+    let batch_inputs = store_client
+        .get_batch_inputs(
+            vec![(block_ref.block_num(), block_ref.hash())].into_iter(),
+            notes.iter().map(Note::id),
```

Comment:
This basically marks every not as unauthenticated, right? And I think we do this so that we can get Merkle paths for every note - correct? If so, I would add some comments here explaining this.